File: ARCADE/Intro/Level 2_5: matrixElementsSum.cs
    using System;
     
    public class Test
    {
    	public static int matrixElementsSum(int[][] matrix){
    	    int i, j;
    	    int count = 0;
    	    for (j = 0; j < matrix[0].Length; j++){
    	        for (i = 0; i < matrix.Length; i++){
    	            if (matrix[i][j] == 0) break;
    	            count += matrix[i][j];
    	        }
    	    }
    	    return count;
    	}
    	public static void Main()
    	{
    		int [][] matrix = new int [][]{
    			new int []{0, 1, 1, 2},
    			new int []{0, 5, 0, 0},
    			new int []{2, 0, 3, 3}
    		};
    		Console.WriteLine(matrixElementsSum(matrix));
    	}
    }

File: ARCADE/Intro/Level 2_5: matrixElementsSum.pas
program ideone;
Type 
	ArrayInteger = Array of integer;
	ArrayArrayInteger = Array of ArrayInteger;
 
function matrixElementsSum(matrix: ArrayArrayInteger): integer;
var i, j, count: integer;
begin
    count := 0;
    for j := 0 to length(matrix[0]) -1 do
        for i := 0 to length(matrix) -1 do
            begin
                if(matrix[i][j] = 0) then break;
                count := count + matrix[i][j];
            end;
    matrixElementsSum := count;
end;
 
var matrix : ArrayArrayInteger;
begin
	matrix := ArrayArrayInteger.Create(ArrayInteger.Create(0, 1, 1, 2),
	                                   ArrayInteger.Create(0, 5, 0, 0),
	                                   ArrayInteger.Create(2, 0, 3, 3));
	writeln(matrixElementsSum(matrix));
end.
